doc/ansible/user-guide.md

```diff
@@ -0,0 +1,337 @@
+
+```yaml
+---
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+```
+
+#### Options
+**Role**
+Specifying a `role` option in `watches.yaml` will configure the operator to use
+this specified path when launching `ansible-runner` with an Ansible Role. By
+default, the `new` command will fill in an absolute path to where your role
+should go.
+```yaml
+---
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+  role: /opt/ansible/roles/Memcached
+```
+
+**Playbook**
+Specifying a `playbook` option in `watches.yaml` will configure the operator to
+use this specified path when launching `ansible-runner` with an Ansible
+Playbook
+```yaml
+---
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+  playbook: /opt/ansible/playbook.yaml
+```
+
+## Building the Memcached Ansible Role
+
+The first thing to do is to modify the generated Ansible role under
+`roles/Memcached`. This Ansible Role controls the logic that is executed when a
+resource is modified.
+
+### Define the Memcached spec
+
+Defining the spec for an Ansible Operator can be done entirely in Ansible. The
+Ansible Operator will simply pass all key value pairs listed in the Custom
+Resource spec field along to Ansible as
+[variables](https://docs.ansible.com/ansible/2.5/user_guide/playbooks_variables.html#passing-variables-on-the-command-line).
+It is recommended that you perform some type validation in Ansible on the
+variables to ensure that your application is receiving expected input.
+
+First, set a default in case the user doesn't set the `spec` field by modifying
+`roles/Memcached/defaults/main.yml`:
+```yaml
+size: 1
+```
+
+### Defining the Memcached deployment
+
+Now that we have the spec defined, we can define what Ansible is actually
+executed on resource changes. Since this is an Ansible Role, the default
+behavior will be to execute the tasks in `roles/Memcached/tasks/main.yml`. We
+want Ansible to create a deployment if it does not exist which runs the
+`memcached:1.4.36-alpine` image. Ansible 2.5+ supports the [k8s Ansible
+Module](https://docs.ansible.com/ansible/2.6/modules/k8s_module.html) which we
+will leverage to control the deployment definition.
+
+Modify `roles/Memcached/tasks/main.yml` to look like the following:
+```yaml
+---
+- name: start memcached
+  k8s:
+    definition:
+      kind: Deployment
+      apiVersion: apps/v1
+      metadata:
+        name: '{{ meta.name }}-memcached'
+        namespace: '{{ meta.namespace }}'
+      spec:
+        replicas: "{{size}}"
+        selector:
+          matchLabels:
+            app: memcached
+        template:
+          metadata:
+            labels:
+              app: memcached
+          spec:
+            containers:
+            - name: memcached
+              command:
+              - memcached
+              - -m=64
+              - -o
+              - modern
+              - -v
+              image: "memcached:1.4.36-alpine"
```

Comment:
Sorry, one more thing. This should be fully qualified so it will work with non-docker runtimes. I think just prefixing it with `docker.io/` will work.

Reply:
Ah thanks.